### PR TITLE
Extension UI implementation

### PR DESCRIPTION
**Pull request by nidhitalastha**
This pull request introduces several new components and refactors existing ones to improve the chat interface in the `webview-ui` project. The changes include adding new components for the header, chat input, and messages, as well as updating the layout to integrate these components.

### New Components:

* **Header**: Added a new `Header` component to display the application name. (`webview-ui/src/components/Header.tsx`)
* **ChatInput**: Introduced a `ChatInput` component to handle user input and form submission. (`webview-ui/src/components/chat/ChatInput.tsx`)
* **MessageContainer**: Created a `MessageContainer` component to encapsulate the structure of messages, differentiating between system and user messages. (`webview-ui/src/components/chat/MessageContainer.tsx`)
* **SystemMessage**: Added a `SystemMessage` component to display system-generated messages, including actions like running commands. (`webview-ui/src/components/chat/SystemMessage.tsx`)
* **UserMessage**: Added a `UserMessage` component to display user-generated messages. (`webview-ui/src/components/chat/UserMessage.tsx`)

### Layout and Integration:

* **ChatContainer**: Added a `ChatContainer` component to manage and render a series of chat messages using the `SystemMessage` and `UserMessage` components. (`webview-ui/src/components/chat/ChatContainer.tsx`)
* **ChatLayout**: Refactored the `ChatLayout` to integrate the new `Header`, `ChatContainer`, and `ChatInput` components, replacing the previous placeholder elements. (`webview-ui/src/components/chat/Layout.tsx`)

### CODE REVIEW

File: webview-ui/src/components/chat/ChatContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import SystemMessage from "./SystemMessage"
+import UserMessage from "./UserMessage"
+
+
+const demoChat = [
```
Comment (adithya1012) [inline webview-ui/src/components/chat/ChatContainer.tsx:5]: This is the demoChat should be replaced with the dynamic content from User and API while we are integrating.

**Comment by adithya1012** [comment]
I am able to run the extension without the error. The chat option for the user is now not present. We will this feature add in our integrategration meet today.

**Review by adithya1012 — APPROVED** [review]

File: webview-ui/src/components/chat/ChatContainer.tsx
Comment (Sinduja98) [inline webview-ui/src/components/chat/ChatContainer.tsx:1]: Finally while integrating the code, position of terminal needs to be adjusted.

**Review by Sinduja98 — APPROVED** [review]

File: webview-ui/src/components/Header.tsx
Comment (hdprajwal) [inline webview-ui/src/components/Header.tsx:1]: Can we remove the Header implementation and instead use the default extension header component

**Comment by jahnavib26** [comment]
Worked on my system after updating the VSCode version.

**Review by jahnavib26 — APPROVED** [review]

File: webview-ui/src/components/Header.tsx
Comment (nidhitalastha) [reply webview-ui/src/components/Header.tsx:1]: I have removed the header component as suggested.

**Review by hdprajwal — APPROVED** [review]

**Pull request merged**